library/src/actions/minBytes/minBytes.test.ts

```diff
@@ -0,0 +1,95 @@
+import { describe, expect, test } from 'vitest';
+import { expectActionIssue, expectNoActionIssue } from '../../vitest/index.ts';
+import {
+  minBytes,
+  type MinBytesAction,
+  type MinBytesIssue,
+} from './minBytes.ts';
+
+describe('minBytes', () => {
+  describe('should return action object', () => {
+    const baseAction: Omit<MinBytesAction<string, 5, never>, 'message'> = {
+      kind: 'validation',
+      type: 'min_bytes',
+      expects: '>=5',
+      requirement: 5,
+      async: false,
+      _run: expect.any(Function),
+    };
+
+    test('with undefined message', () => {
+      const action: MinBytesAction<string, 5, undefined> = {
+        ...baseAction,
+        message: undefined,
+      };
+      expect(minBytes(5)).toStrictEqual(action);
+      expect(minBytes(5, undefined)).toStrictEqual(action);
+    });
+
+    test('with string message', () => {
+      expect(minBytes(5, 'message')).toStrictEqual({
+        ...baseAction,
+        message: 'message',
+      } satisfies MinBytesAction<string, 5, string>);
+    });
+
+    test('with function message', () => {
+      const message = () => 'message';
+      expect(minBytes(5, message)).toStrictEqual({
+        ...baseAction,
+        message,
+      } satisfies MinBytesAction<string, 5, typeof message>);
+    });
+  });
+
+  describe('should return dataset without issues', () => {
+    const action = minBytes(5);
+
+    test('for untyped inputs', () => {
+      expect(action._run({ typed: false, value: null }, {})).toStrictEqual({
+        typed: false,
+        value: null,
+      });
+    });
+
+    test('for valid strings', () => {
+      expectNoActionIssue(action, ['12345', '123456', 'foobarbaz123']);
+    });
+
+    test('for UTF-8 characters', () => {
+      const value3 = 'あい'; // in UTF-8, 'あい' is 6 bytes
+      expectNoActionIssue(action, [value3]);
+    });
+  });
+
+  describe('should return dataset with issues', () => {
+    const action = minBytes(5, 'message');
+    const getBytesLength = (value: string) => new TextEncoder().encode(value).length;
```

Comment:
I avoid exporting these internal implementation functions to use in tests; for me, it's a leak that final users don't need to know, but I'm up to changing it.

Reply:
Wouldn't it be better to enter the number we expect directly? The calculation is not really necessary, is it?

Reply:
You are right, I have justd changed for a map.